src/DataStructures/Tensor/Expressions/Normalize.hpp

```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions normalize_with_euclidean_metric and normalize
+
+#pragma once
+
+class DataVector;
+
+#include "DataStructures/Tensor/Expressions/Magnitude.hpp"
+#include "DataStructures/Tensor/Expressions/ScaleBy.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Functions
+ * \brief normalize a rank-1 tensor assuming a Euclidean metric
+ *
+ * \returns the normalized rank-1 tensor
+ *
+ * \remark If you need to store the magnitude of the rank-1 tensor and the
+ * normalized tensor,  it is better to call euclidean_magnitude and scale_by
```

Comment:
I recall a past case (inverses, I think), where we had a similar "better" question and decided that the wrapped code was so simple that having the function wasn't worth the possible confusion.  Perhaps we should do that here too.